### ISSUE TITLE
Segfault with latest git on linux

### ISSUE THREAD

**Issue by Kirtai** [openc2e/openc2e#156]
Got this on running latest git.
Log is from running under gdb

```
Starting program: /zpool/media/games/code/engines/openc2e/build/openc2e -d /zpool/games/docking-station/DS/
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
openc2e (development build), built Jun  9 2020 07:47:32
Copyright (c) 2004-2008 Alyssa Milburn and others

Warning: No gametype specified, assuming C3/DS (c3), see --help if you need to specify one.
* Reading catalogue files...
* Initial setup...
* Reading PRAY files...
* Initialising backend qt...
* Initialising audio backend sdlmixer...
[New Thread 0x7fffe2d4e700 (LWP 27937)]
[New Thread 0x7fffe2d0d700 (LWP 27938)]
Listening for connections on port 20001.
* Executing initial scripts...

Thread 1 "openc2e" received signal SIGSEGV, Segmentation fault.
0x00007ffff6745f5b in _IO_new_fclose (fp=0x0) at iofclose.c:48
48      iofclose.c: No such file or directory.
(gdb) bt
#0  0x00007ffff6745f5b in _IO_new_fclose (fp=0x0) at iofclose.c:48
#1  0x0000555555849f3f in mmapifstream::mmapifstream(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >) (this=0x7fffffffb350, filename=..., __in_chrg=<optimised out>, __vtt_parm=<optimised out>)
    at /zpool/media/games/code/engines/openc2e/src/mmapifstream.cpp:50
#2  0x00005555557699fa in tryOpen(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, filetype) (fname=..., ft=s16) at /usr/include/c++/9/bits/basic_string.h:936
#3  0x000055555576a1b0 in imageManager::getImage(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, bool) (this=this@entry=0x5555559b0230, name="blnk", is_background=is_background@entry=false)
    at /usr/include/c++/9/bits/basic_string.h:320
#4  0x000055555572246e in SpritePart::SpritePart(Agent*, unsigned int, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, unsigned int, int, int, unsigned int)
    (this=0x555555c24c30, p=<optimised out>, _id=<optimised out>, spritefile=..., fimg=0, _x=<optimised out>, _y=0, _z=0) at /usr/include/c++/9/bits/basic_string.h:936
#5  0x000055555572265a in DullPart::DullPart(Agent*, unsigned int, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, unsigned int, int, int, unsigned int) (this=0x555555c24c30, p=
    0x555555c58d10, _id=0, spritefile=..., fimg=0, _x=0, _y=0, _z=0) at /usr/include/c++/9/bits/basic_string.h:936
#6  0x000055555578fda7 in SimpleAgent::SimpleAgent(unsigned char, unsigned char, unsigned short, unsigned int, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, unsigned int, unsigned int)
    (this=0x555555c58d10, family=<optimised out>, genus=<optimised out>, species=<optimised out>, plane=<optimised out>, spritefile=..., firstimage=0, imagecount=1) at /usr/include/c++/9/bits/char_traits.h:300
#7  0x0000555555707572 in caosVM::c_NEW_SIMP() (this=0x7fffffffc2b0) at /usr/include/c++/9/bits/basic_string.h:936
#8  0x00005555557176e8 in caosVM::invoke_cmd(script*, bool, int) (this=0x7fffffffc2b0, s=s@entry=0x555555992270, is_saver=is_saver@entry=false, opidx=opidx@entry=179)
    at /zpool/media/games/code/engines/openc2e/src/caosVM.cpp:109
#9  0x0000555555718005 in caosVM::runOpCore(script*, caosOp) (this=this@entry=0x7fffffffc2b0, s=s@entry=0x555555992270, op=..., op@entry=...) at /zpool/media/games/code/engines/openc2e/src/caosVM.cpp:157
#10 0x000055555571980c in caosVM::runOp() (this=0x7fffffffc2b0) at /zpool/media/games/code/engines/openc2e/src/caosVM.cpp:330
#11 0x0000555555716599 in caosVM::runEntirely(std::shared_ptr<script>) (this=this@entry=0x7fffffffc2b0, s=std::shared_ptr<class script> (use count 4, weak count 0) = {...})
    at /zpool/media/games/code/engines/openc2e/src/caosVM.cpp:354
#12 0x000055555579d762 in World::executeInitScript(ghc::filesystem::path) (this=<optimised out>, p=...) at /usr/include/c++/9/ext/atomicity.h:96
#13 0x000055555579ee2e in World::executeBootstrap(ghc::filesystem::path) (this=0x55555598f500 <world>, p=...) at /usr/include/c++/9/bits/basic_string.h:936
#14 0x000055555579f852 in World::executeBootstrap(bool) (this=0x55555598f500 <world>, switcher=switcher@entry=false) at /usr/include/c++/9/bits/basic_string.h:936
#15 0x0000555555738d6e in Engine::initialSetup() (this=<optimised out>) at /zpool/media/games/code/engines/openc2e/src/Engine.cpp:982
#16 0x00005555556b0551 in main(int, char**) (argc=3, argv=0x7fffffffdf78) at /zpool/media/games/code/engines/openc2e/src/main.cpp:66
(gdb) 
```

**Comment by ligfx** [comment]
Should be fixed in 7698512 - @Kirtai can you confirm?

**Comment by Kirtai** [comment]
Yep, builds and runs fine with latest git.
